From 2a0cfeee13e1c1aafe33b3a6c2fadc36cbd30485 Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:49 -0500
Subject: [wireless] iwlwifi/mvm: update values for Smart Fifo

Message-id: <1418207389-7899-36-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101607
O-Subject: [RHEL7.1 PATCH 35/35] iwlwifi: mvm: update values for Smart Fifo
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

BZ#1155538

Backport from wireless-testing of:

commit b4c82adcba8cb4b23068a6b800ca98da3bee6888
Author: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Date:   Mon Dec 1 16:44:09 2014 +0200

    iwlwifi: mvm: update values for Smart Fifo

    Interoperability issues were identified and root caused to
    the Smart Fifo watermarks. These issues arose with
    NetGear R7000. Fix this.

    Cc: <stable@vger.kernel.org> [3.14+]
    Fixes: 1f3b0ff8ecce ("iwlwifi: mvm: Add Smart FIFO support")
    Reviewed-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Tested on iwl7265 adapter.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/wireless/iwlwifi/mvm/fw-api.h b/drivers/net/wireless/iwlwifi/mvm/fw-api.h
index 41a1816..bafdf24 100644
--- a/drivers/net/wireless/iwlwifi/mvm/fw-api.h
+++ b/drivers/net/wireless/iwlwifi/mvm/fw-api.h
@@ -1466,7 +1466,7 @@ enum iwl_sf_scenario {
 #define SF_NUM_TIMEOUT_TYPES 2  /* Aging timer and Idle timer */
 
 /* smart FIFO default values */
-#define SF_W_MARK_SISO 4096
+#define SF_W_MARK_SISO 6144
 #define SF_W_MARK_MIMO2 8192
 #define SF_W_MARK_MIMO3 6144
 #define SF_W_MARK_LEGACY 4096
-- 
1.7.1